lib/ansible/modules/cloud/hcloud/hcloud_ssh_key.py

```diff
@@ -0,0 +1,219 @@
+    type: dict
+    sample: {
+        "id": 1937415,
+        "name": "mein-ssh_key",
+        "fingerprint": "b7:2f:30:a0:2f:6c:58:6c:21:04:58:61:ba:06:3b:2f",
+        "public_key": "ssh-rsa AAAjjk76kgf...Xt",
+        "labels":{}
+    }
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.hcloud import Hcloud
+
+try:
+    from hcloud.volumes.domain import Volume
+    from hcloud.ssh_keys.domain import SSHKey
+    from hcloud.ssh_keys.domain import Server
+    from hcloud import APIException
+except ImportError:
+    pass
+
+
+class AnsibleHcloudSSHKey(Hcloud):
+    def __init__(self, module):
+        Hcloud.__init__(self, module, "hcloud_ssh_key")
+        self.hcloud_ssh_key = None
+
+    def _prepare_result(self):
+        return {
+            "id": to_native(self.hcloud_ssh_key.id),
+            "name": to_native(self.hcloud_ssh_key.name),
+            "fingerprint": to_native(self.hcloud_ssh_key.fingerprint),
+            "public_key": to_native(self.hcloud_ssh_key.public_key),
+            "labels": self.hcloud_ssh_key.labels,
+        }
+
+    def _get_ssh_key(self):
+        try:
+            if self.module.params.get("id") is not None:
+                self.hcloud_ssh_key = self.client.ssh_keys.get_by_id(
+                    self.module.params.get("id")
+                )
+            elif self.module.params.get("name") is not None:
+                self.hcloud_ssh_key = self.client.ssh_keys.get_by_name(
+                    self.module.params.get("name")
+                )
+            elif self.module.params.get("fingerprint") is not None:
+                self.hcloud_ssh_key = self.client.ssh_keys.get_by_fingerprint(
+                    self.module.params.get("fingerprint")
+                )
+
+        except APIException as e:
+            self.module.fail_json(msg=e.message)
+
+    def _create_ssh_key(self):
+        self.module.fail_on_missing_params(
+            required_params=["name", "public_key"]
+        )
+        params = {
+            "name": self.module.params.get("name"),
+            "public_key": self.module.params.get("public_key"),
+            "labels": self.module.params.get("labels")
+        }
+
+        if not self.module.check_mode:
+            self.client.ssh_keys.create(**params)
+        self._mark_as_changed()
+        self._get_ssh_key()
+
+    def _update_ssh_key(self):
+        name = self.module.params.get("name")
+        if name is not None and self.hcloud_ssh_key.name != name:
+            self.module.fail_on_missing_params(
+                required_params=["id"]
+            )
+            if not self.module.check_mode:
+                self.hcloud_ssh_key.update(name)
+            self._mark_as_changed()
+
+        labels = self.module.params.get("labels")
+        if labels is not None:
+            if not self.module.check_mode:
+                self.hcloud_ssh_key.update(labels)
```

Comment:
I realized this would not be idempotent and because this seems to be an issue in all modules, we should handle the "does the labels need an update" in module_utils.

Reply:
This is indented.

On our platform you need to specify all labels when you update the labels. 
https://docs.hetzner.cloud/#ssh-keys-update-an-ssh-key

We want to have the same user experience on all plattforms.

Reply:
This is fine, but we should only update (all labels) if one changed, misses or should be added.

Reply:
I have added a solution and tests for it.